object_store/src/lib.rs

```diff
@@ -850,6 +850,13 @@ impl GetResult {
     }
 }
 
+/// Result for a put request
+#[derive(Debug)]
```

Comment:
I always find it annoying when these traits are not derived for simple objects. 

```suggestion
#[derive(Debug, Hash, Clone, PartialOrd)]
```

Reply:
I've added Clone, but I'm not sure about Hash and PartialOrd here

Reply:
Clone is probably good enough to start with